### PR TITLE
Enable unpacking combined factorizations

### PR DESCRIPTION
**Pull request by upsj**
This implements the `Factorization::unpack` function for LU and Cholesky factorizations. There are no new kernels, so I only test the core functionality.

### CODE REVIEW

**Review by yhmtsai — APPROVED** [review]
also need to check the composition and symm composition unpack

File: core/factorization/factorization.cpp [hunk 1]
```diff
@@ -33,18 +33,88 @@ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
...
+Factorization<ValueType, IndexType>::unpack() const
+{
+    const auto exec = this->get_executor();
+    const auto size = this->get_size();
+    switch (this->get_storage_type()) {
+    case storage_type::empty:
+        GKO_NOT_SUPPORTED(nullptr);
+    case storage_type::composition:
+    case storage_type::symm_composition:
+        return this->clone();
```
Comment (pratikvn) [inline core/factorization/factorization.cpp:72]: Why return a clone for the symmetric version here ?

**Review by pratikvn — APPROVED** [review]

File: core/factorization/factorization.cpp (hunk 1 above)
Comment (upsj) [reply core/factorization/factorization.cpp:72]: the function returns a unique_ptr, we cannot fulfill that requirement from `this`

Comment (upsj) [reply core/factorization/factorization.cpp:72]: should probably add a note to the documentation on this.

Comment (pratikvn) [reply core/factorization/factorization.cpp:72]: And that is not the case for a non-symmetric composition ?

Comment (upsj) [reply core/factorization/factorization.cpp:72]: this is a fallthrough, both `composition` and `symm_composition` lead to the statement being executed.

**Pull request merged**